ql/src/java/org/apache/hadoop/hive/ql/metadata/VirtualColumn.java

```diff
@@ -50,10 +50,14 @@
 
   RAWDATASIZE("RAW__DATA__SIZE", TypeInfoFactory.longTypeInfo),
   /**
-   * {@link org.apache.hadoop.hive.ql.io.RecordIdentifier} 
+   * {@link org.apache.hadoop.hive.ql.io.RecordIdentifier}
    */
   ROWID("ROW__ID", RecordIdentifier.StructInfo.typeInfo, true, RecordIdentifier.StructInfo.oi),
   ROWISDELETED("ROW__IS__DELETED", TypeInfoFactory.booleanTypeInfo),
+  PARTITION_SPEC_ID("PARTITION__SPEC__ID", TypeInfoFactory.intTypeInfo),
+  PARTITION_HASH("PARTITION__HASH", TypeInfoFactory.longTypeInfo),
+  FILE_PATH("FILE__PATH", TypeInfoFactory.stringTypeInfo),
+  ROW_POSITION("ROW__POSITION", TypeInfoFactory.longTypeInfo),
```

Comment:
How is this handled inside the `ROW__ID`?

Reply:
The row number is obtained in the `OrcRecordUpdater`:
```
    ByteBuffer val =
        reader.getMetadataValue(OrcRecordUpdater.ACID_KEY_INDEX_NAME)
```
then the stripe metadata is deserialized and this contains the row number